src/hardware/siglent-sdl10x0/api.c

```diff
@@ -0,0 +1,461 @@
+		ivalue = g_variant_get_double(data);
+		g_snprintf(command, sizeof(command), ":CURR:PROT:LEV %.3f", (ivalue));
+		sr_spew("Sending '%s'.", command);
+		ret = sr_scpi_send(sdi->conn, command);
+		break;
+	default:
+		ret = SR_ERR_NA;
+	}
+
+	return ret;
+}
+
+static int config_list(uint32_t key, GVariant **data,
+	const struct sr_dev_inst *sdi, const struct sr_channel_group *cg)
+{
+
+	(void)data;
+	struct dev_context *devc;
+
+	devc = sdi ? sdi->priv : NULL;
+
+	if (!cg) {
+		return STD_CONFIG_LIST(key, data, sdi, cg, scanopts, drvopts, devopts);
+	} else {
+		switch (key) {
+		case SR_CONF_DEVICE_OPTIONS:
+			*data = std_gvar_array_u32(ARRAY_AND_SIZE(devopts_cg));
+			break;
+		case SR_CONF_REGULATION:
+			*data = std_gvar_array_str(ARRAY_AND_SIZE(regulation));
+			break;
+		case SR_CONF_VOLTAGE_TARGET:
+			*data = std_gvar_min_max_step(0.0, 150.0, 0.001);
+			break;
+		case SR_CONF_CURRENT_LIMIT:
+			*data = std_gvar_min_max_step(0.0, 30.0, 0.001);
+			break;
+		case SR_CONF_POWER_TARGET:
+			if (!devc) {
+				*data = std_gvar_min_max_step(0.0, 200.0, 0.001);
+			} else {
+				*data = std_gvar_min_max_step(0.0, devc->maxpower, 0.001);
+			}
+			break;
+		case SR_CONF_RESISTANCE_TARGET:
+			*data = std_gvar_min_max_step(0.03, 10000.0, 0.01);
+			break;
+		case SR_CONF_OVER_POWER_PROTECTION_THRESHOLD:
+			if (!devc) {
+				*data = std_gvar_min_max_step(0.0, 200.0, 0.001);
+			} else {
+				*data = std_gvar_min_max_step(0.0, devc->maxpower, 0.001);
+			}
+			break;
+		case SR_CONF_OVER_CURRENT_PROTECTION_THRESHOLD:
+			*data = std_gvar_min_max_step(0.0, 30.0, 0.001);
+			break;
+		default:
+			return SR_ERR_NA;
+		}
+	}
+	return SR_OK;
+}
+
+static int dev_acquisition_start(const struct sr_dev_inst *sdi)
+{
+	struct sr_scpi_dev_inst *scpi;
+	struct dev_context *devc;
+
+	scpi = sdi->conn;
+	devc = sdi->priv;
+
+	sr_sw_limits_acquisition_start(&devc->limits);
+	std_session_send_df_header(sdi);
+
+	/*
+	 * Start acquisition. The receive routine will continue
+	 * driving the acquisition.
+	 */
+	sr_scpi_send(scpi, "MEAS:VOLT?");
+	devc->acq_state = ACQ_REQUESTED_VOLTAGE;
+	return sr_scpi_source_add(sdi->session, scpi, G_IO_IN, 100, siglent_sdl10x0_handle_events, (void *)sdi);
+}
+
+static int dev_acquisition_stop(struct sr_dev_inst *sdi)
+{
+	struct sr_scpi_dev_inst *scpi;
+
+	scpi = sdi->conn;
+
+	sr_scpi_source_remove(sdi->session, scpi);
+	std_session_send_df_end(sdi);
+
+	return SR_OK;
+}
+
+static struct sr_dev_driver siglent_sdl10x0_driver_info = {
+	.name = "siglent-sdl10x0",
+	.longname = "SIGLENT SDL10x0",
+	.api_version = 1,
+	.init = std_init,
+	.cleanup = std_cleanup,
+	.scan = scan,
+	.dev_list = std_dev_list,
+	.dev_clear = std_dev_clear,
+	.config_get = config_get,
+	.config_set = config_set,
+	.config_list = config_list,
+	.dev_open = dev_open,
+	.dev_close = dev_close,
+	.dev_acquisition_start = dev_acquisition_start,
+	.dev_acquisition_stop = dev_acquisition_stop,
+	.context = NULL,
```

Comment:
No need to fill NULL entries.

Reply:
I found [a](https://github.com/sigrokproject/libsigrok/blob/0dfa850efb5553940982633ab96a8612860e87b4/src/hardware/korad-kaxxxxp/api.c#L546) [number](https://github.com/sigrokproject/libsigrok/blob/0dfa850efb5553940982633ab96a8612860e87b4/src/hardware/maynuo-m97/api.c#L428) [of](https://github.com/sigrokproject/libsigrok/blob/0dfa850efb5553940982633ab96a8612860e87b4/src/hardware/mastech-ms6514/api.c#L215) [examples](https://github.com/sigrokproject/libsigrok/blob/0dfa850efb5553940982633ab96a8612860e87b4/src/hardware/manson-hcs-3xxx/api.c#L351) where this was still in the driver, so I left it in for consistency. I can remove it if  you wish, though.

Reply:
Yeah you're right, and I see it's in the new-driver template as well. Never mind, it needs a patch to fix up all of them.